packages/engine-core/src/data-proxy/DataProxyEngine.ts

```diff
@@ -91,6 +192,48 @@ export class DataProxyEngine extends Engine<DataProxyTxInfoPayload> {
   async start() {}
   async stop() {}
 
+  private propagateResponseExtensions(extensions: DataProxyExtensions): void {
+    const tracingConfig = getTracingConfig(this.config.previewFeatures || [])
+
+    if (extensions?.logs?.length) {
+      extensions.logs.forEach((log) => {
+        switch (log.level) {
+          case 'debug':
+          case 'error':
+          case 'trace':
+          case 'warn':
+          case 'info':
+            // TODO these are propgated into the response.errors key
+            break
+          case 'query': {
+            let dbQuery =
+              typeof log.attributes?.query === 'string' && log.attributes?.query ? log.attributes.query : log.name
+            if (!tracingConfig.enabled) {
+              // The engine uses tracing to consolidate logs
+              //  - and so we should strip the generated traceparent
+              //  - if tracing is disabled.
+              // Example query: 'SELECT /* traceparent=00-123-0-01 */'
+              const [query] = dbQuery.split('/* traceparent')
+              dbQuery = query
+            }
+
+            this.logEmitter.emit('query', {
+              query: dbQuery,
+              timestamp: log.timestamp,
+              duration: log.attributes.duration_ms,
+              // params: log.params - Missing
+              // target: log.target - Missing
```

Comment:
They are both `string` in the user-facing `QueryEvent` type, should we at least pass empty strings here instead?

Reply:
I think they are present now, gonna address.